features/step_definitions/name_page_steps.rb

```diff
@@ -0,0 +1,7 @@
+Given(/^When I visit the name page for "([^"]*)"$/) do |name|
+  visit '/name/Amber'
+end
+
+Then(/^I should create a new user with name "([^"]*)"$/) do |name|
+  User.create(name: params[:name])
```

Comment:
You should not create a User in this step definition but rather assert (using `expect`) that there is a User with a specific name. Also, in step definitions, the `params` hash is not available.